src/game/client/components/broadcast.cpp

```diff
@@ -80,21 +100,41 @@ void CBroadcast::OnMessage(int MsgType, void *pRawMsg)
 	}
 }
 
-void CBroadcast::DoBroadcast(const char *pText)
+void CBroadcast::DeleteBroadcastContainer()
 {
-	str_copy(m_aBroadcastText, pText);
-	m_BroadcastTick = Client()->GameTick(g_Config.m_ClDummy) + Client()->GameTickSpeed() * 10;
 	m_BroadcastRenderOffset = -1.0f;
 	TextRender()->DeleteTextContainer(m_TextContainerIndex);
+}
+
+void CBroadcast::DoDummyBroadcast(const char *pText)
+{
+	DoBroadcast(pText, true);
+}
+
+void CBroadcast::DoBroadcast(const char *pText, bool Dummy)
+{
+	int Conn = (Dummy != (bool)g_Config.m_ClDummy);
+	m_aBroadcastTick[Conn] = Client()->GameTick(g_Config.m_ClDummy) + Client()->GameTickSpeed() * 10;
+	str_copy(m_aaBroadcastText[Conn], pText);
+
+	if(Dummy)
+	{
+		const float SecondsRemaining = (m_aBroadcastTick[Conn] - Client()->GameTick(g_Config.m_ClDummy)) / (float)Client()->GameTickSpeed();
+		if(SecondsRemaining <= 0.0f)
+			DeleteBroadcastContainer();
+	}
+	else
+		DeleteBroadcastContainer();
 
 	if(g_Config.m_ClPrintBroadcasts)
 	{
-		char aLine[sizeof(m_aBroadcastText)];
+		char aLine[sizeof(m_aaBroadcastText[0])];
 		while((pText = str_next_token(pText, "\n", aLine, sizeof(aLine))))
 		{
 			if(aLine[0] != '\0')
 			{
-				GameClient()->Console()->Print(IConsole::OUTPUT_LEVEL_STANDARD, "broadcast", aLine, color_cast<ColorRGBA>(ColorHSLA(g_Config.m_ClMessageHighlightColor)));
+				log_info_color(LOG_COLOR::ColorToLogColor(color_cast<ColorRGBA>(ColorHSLA(g_Config.m_ClMessageHighlightColor))),
+					Dummy ? "dummy broadcast" : "broadcast", aLine);
```

Comment:
```suggestion
					Dummy ? "dummy broadcast" : "broadcast", "%s", aLine);
```

Let's not add security vulnerabilities here 

<img width="70" height="70" alt="Image" src="https://github.com/user-attachments/assets/10d914e9-ecb2-4a01-97ec-3633dcb290ca" />